srai/loaders/osm_way_loader/osm_way_loader.py

```diff
@@ -0,0 +1,291 @@
+            .distinct()
+            .to_list()
+        )
+        self.osm_keys = list(osm_way_tags.keys())
+
+    def load(self, area: gpd.GeoDataFrame) -> Tuple[gpd.GeoDataFrame, gpd.GeoDataFrame]:
+        """
+        Load road infrastructure for a given GeoDataFrame.
+
+        Args:
+            area (gpd.GeoDataFrame): (Multi)Polygons for which to download road infrastructure data.
+
+        Raises:
+            ValueError: If provided GeoDataFrame has no crs defined.
+
+        Returns:
+            Tuple[gpd.GeoDataFrame, gpd.GeoDataFrame]: Road infrastructure as (nodes, edges).
+        """
+        import osmnx as ox
+
+        ox.settings.useful_tags_way = constants.OSMNX_WAY_KEYS
+        ox.settings.timeout = constants.OSMNX_TIMEOUT
+
+        gdf_wgs84 = area.to_crs(crs=srai_constants.WGS84_CRS)
+
+        gdf_nodes_raw, gdf_edges_raw = self._gdfs_from_polygons(gdf_wgs84)
+        gdf_edges = self._explode_cols(gdf_edges_raw)
+
+        if self.preprocess:
+            self._preprocess(gdf_edges, inplace=True)
+
+        if self.wide:
+            gdf_edges = self._to_wide(gdf_edges_raw, gdf_edges)
+
+        return gdf_nodes_raw, gdf_edges
+
+    def _gdfs_from_polygons(
+        self, gdf: gpd.GeoDataFrame
+    ) -> Tuple[gpd.GeoDataFrame, gpd.GeoDataFrame]:
+        """
+        Obtain the raw road infrastructure data from OSM.
+
+        Args:
+            gdf (gpd.GeoDataFrame): (Multi)Polygons for which to download road infrastructure data.
+
+        Notes:
+            * The road infrastructure graph is treated as an undirected graph.
+            * `FIXME` The result may contain duplicated nodes or edges,
+            when nearby polygons are queried.
+
+        Returns:
+            Tuple[gpd.GeoDataFrame, gpd.GeoDataFrame]: Road infrastructure as (nodes, edges).
+        """
+        import osmnx as ox
+
+        nodes = []
+        edges = []
+        for polygon in tqdm(gdf["geometry"], desc="Downloading graphs", leave=False):
+            G_directed = ox.graph_from_polygon(
+                polygon, network_type=self.network_type, retain_all=True, clean_periphery=True
+            )
+
+            G_undirected = ox.utils_graph.get_undirected(G_directed)
+            gdf_n, gdf_e = ox.graph_to_gdfs(G_undirected)
+            nodes.append(gdf_n)
+            edges.append(gdf_e)
+
+        gdf_nodes = pd.concat(nodes, axis=0)
+        gdf_edges = pd.concat(edges, axis=0)
+
+        return gdf_nodes, gdf_edges
+
+    def _explode_cols(self, gdf: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
+        """
+        Explode lists in feature columns.
+
+        Args:
+            gdf (gpd.GeoDataFrame): Edges with columns to explode.
+
+        Returns:
+            gpd.GeoDataFrame: Edges with all of their columns exploded.
+        """
+        for col in self.osm_keys:
+            gdf = gdf.explode(col)
+
+        gdf["i"] = range(0, len(gdf))
+        gdf.set_index("i", append=True, inplace=True)
```

Comment:
Why this instead of `.reset_index()`?

Reply:
What I want to do here is to append an ascending number to the existing index, not just get rid of the old index altogether

Reply:
Makes sense, thanks ;)